conans/client/cmd/uploader.py

```diff
@@ -583,14 +583,14 @@ def compress_files(files, symlinks, name, dest_dir, output=None):
                                                      "Compressing %s" % name) as pg_file_list:
             for filename, abs_path in pg_file_list:
                 info = tarfile.TarInfo(name=filename)
-                info.size = os.stat(abs_path).st_size
-                info.mode = os.stat(abs_path).st_mode & mask
                 if os.path.islink(abs_path):
                     info.type = tarfile.SYMTYPE
                     info.size = 0  # A symlink shouldn't have size
                     info.linkname = os.readlink(abs_path)  # @UndefinedVariable
                     tgz.addfile(tarinfo=info)
                 else:
+                    info.mode = os.stat(abs_path).st_mode & mask
```

Comment:
wouldn't symlinks suffer the same issue of regular files regarding permissions/modes? (this mask was added because docker). Should use same mask but with ``os.lstat()``?
It would also be nice to avoid calling ``os.stat()`` twice for the same file.